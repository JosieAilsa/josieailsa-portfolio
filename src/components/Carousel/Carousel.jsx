import Button from "../Button/Button";
import Arrow from "../../assets/images/icons/arrow.svg"
import "./Carousel.scss"
import CarouselItem from "../CarouselItem/CarouselItem";
import {projects} from "../../assets/data/data"
import { useState } from "react";

const Carousel = () => {
    const [currentProjectShown, setCurrentProjectShown] = useState(0)
    console.log(currentProjectShown)
    const projectsCarouselJSX = projects.map((project, index) => {
        return <CarouselItem key = {index} image = {project.image} number={index} title = {project.projectTitile} description = {project.description}/>
    })

    const onRightClick = () => {
        console.log("red")
        setCurrentProjectShown(currentProjectShown + 1)
    }

    return (
    <section className="carousel">
        <div className="carousel__container">
            {projectsCarouselJSX[currentProjectShown]}
        </div>
        <div className="carousel__buttons">
            <Button classNames="carousel__button" content={<img src={Arrow} alt="" />}/>
            <Button classNames="carousel__button" onClick = {onRightClick} content={<img src={Arrow} alt="" />}/>    
        </div>
    </section> 
     );
}
 
export default Carousel;