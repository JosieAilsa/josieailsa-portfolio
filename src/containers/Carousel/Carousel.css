@import url("https://fonts.googleapis.com/css2?family=Montserrat:wght@300;400&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Playfair+Display:ital,wght@0,400;0,500;0,700;1,500;1,700;1,765&display=swap");
.carousel {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.carousel .carousel__image-container {
  gap: 1rem;
  display: flex;
  justify-content: flex-start;
}
.carousel .carousel-item__copy {
  width: 100%;
  padding-left: 7%;
}
.carousel .carousel-item__copy .carousel-item__header {
  width: 100%;
  height: 20%;
}
.carousel .carousel-item__copy .carousel-item__text {
  width: 100%;
}