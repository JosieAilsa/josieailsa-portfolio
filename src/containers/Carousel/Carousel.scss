@use "../../assets/sass/abstracts/mixin" as *;
@use "../../assets/sass/abstracts/variables" as *;

.carousel{
    @include flex-center-column;
    gap: 1rem;
    .carousel__container{
        @include flex-center-row;
    }
}