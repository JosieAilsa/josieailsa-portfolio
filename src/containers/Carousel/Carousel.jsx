import Button from "../../components/Button/Button";
import Arrow from "../../assets/images/icons/arrow.svg"
import "./Carousel.scss"
import CarouselItem from "../../components/CarouselItem/CarouselItem";
import {projects} from "../../assets/data/data"
import { useState } from "react";

const Carousel = () => {
    const [currentProjectShown, setCurrentProjectShown] = useState(0)
    console.log(currentProjectShown)
    const projectsCarouselJSX = projects.map((project, index) => {
        return <CarouselItem key = {index} image = {project.image} number={index} title = {project.projectTitile} description = {project.description}/>
    })

    const onClick = (e) => {
        if(e.target.value === "click-right"){
            currentProjectShown === 4 ? setCurrentProjectShown(0):setCurrentProjectShown(currentProjectShown + 1)
            return 
        }
        console.log()
        currentProjectShown === 0 ? setCurrentProjectShown(4):setCurrentProjectShown(currentProjectShown - 1)
        
    }

    return (
    <section className="carousel">
        <div className="carousel__container">
            {projectsCarouselJSX[currentProjectShown]}
        </div>
        <div className="carousel__buttons">
            <Button classNames="carousel__button" value="click-left" onClick = {onClick}  content={<img src={Arrow} alt="" />}/>
            <Button classNames="carousel__button" value="click-right" onClick = {onClick} content={<img src={Arrow} alt="" />}/>    
        </div>
    </section> 
     );
}
 
export default Carousel;