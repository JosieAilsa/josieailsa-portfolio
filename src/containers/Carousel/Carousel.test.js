import { customRender } from "../../utils/testutils";
import Carousel from "./Carousel";

test("Carousel should render", () => {
    const { container } = customRender(<Carousel/>);
    expect(container).toMatchSnapshot();
  });
  