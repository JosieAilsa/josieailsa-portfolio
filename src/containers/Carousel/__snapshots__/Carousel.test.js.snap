// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should render 1`] = `
<div>
  <section
    class="carousel"
  >
    <div
      class="carousel__buttons"
    >
      <button
        class="carousel__button"
        role="button"
      >
        <img
          alt=""
          src="arrow.svg"
        />
      </button>
      <button
        class="carousel__button"
        role="button"
      >
        <img
          alt=""
          src="arrow.svg"
        />
      </button>
    </div>
    <div
      class="carousel__copy"
    >
      <h3
        class="carousel__header"
      >
        Project Name
      </h3>
      <p
        class="carousel__text"
      >
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Quia possimus minima maiores eos iusto quam ut exercitationem sit assumenda, corporis cum incidunt aliquam facere placeat, qui aliquid quod esse deleniti.
      </p>
    </div>
  </section>
</div>
`;
